components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/constant/SQLConstants.java

```diff
@@ -86,9 +86,11 @@
             "(NAME), " +
             "LAST_MODIFIED = VALUES(LAST_MODIFIED), HAS_FILE = VALUES(HAS_FILE), HAS_ATTRIBUTE = VALUES" +
             "(HAS_ATTRIBUTE), TYPE_ID = VALUES(TYPE_ID)";
+    public static final String UPDATE_RESOURCE =
+            "UPDATE IDN_CONFIG_RESOURCE SET LAST_MODIFIED = ?, HAS_FILE = ?, HAS_ATTRIBUTE = ? WHERE ID = ?";
     public static final String UPDATE_RESOURCE_H2 =
-            "UPDATE IDN_CONFIG_RESOURCE SET ID = ?, TENANT_ID = ?, NAME = ?, LAST_MODIFIED = ?, HAS_FILE = ?, " +
```

Comment:
is the old query where we used to allow the id to be changed used by existing features? if so, let's add a new query without changing existing one

Reply:
I have already added a new one for the new functionality required. But this query is wrong. In this table, the ID is unique. Here the query is updating all the values without any validation. Since there's no validation, all the entries will get updated giving the unique key constraint violation.